protocols/virtualfund/virtualfund_single_hop_test.go

```diff
@@ -219,80 +218,32 @@ func TestCrankAsAlice(t *testing.T) {
 	p := consensus_channel.NewAddProposal(o.ToMyRight.Channel.Id, 2, o.ToMyRight.getExpectedGuarantee(), big.NewInt(6))
 	sp := consensus_channel.SignedProposal{Proposal: p}
 	assertProposalSent(t, got, sp, p1)
-	ok(t, err)
-	equals(t, waitingFor, WaitingForCompleteFunding)
+	helpers.Ok(t, err)
+	helpers.Equals(t, waitingFor, WaitingForCompleteFunding)
 
 	// Check idempotency
 	emptySideEffects := protocols.SideEffects{}
 	oObj, got, waitingFor, err = o.Crank(&my.privateKey)
 	o = oObj.(*Objective)
-	ok(t, err)
-	equals(t, got, emptySideEffects)
-	equals(t, waitingFor, WaitingForCompleteFunding)
+	helpers.Ok(t, err)
+	helpers.Equals(t, got, emptySideEffects)
+	helpers.Equals(t, waitingFor, WaitingForCompleteFunding)
 }
 
-// Copied from https://github.com/benbjohnson/testing
-
-// makeRed sets the colour to red when printed
-const makeRed = "\033[31m"
-
-// makeBlack sets the colour to black when printed.
-// as it is intended to be used at the end of a string, it also adds two linebreaks
-const makeBlack = "\033[39m\n\n"
-
-// assert fails the test if the condition is false.
-func assert(tb testing.TB, condition bool, msg string, v ...interface{}) {
-	if !condition {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf(makeRed+"%s:%d: "+msg+makeBlack, append([]interface{}{filepath.Base(file), line}, v...)...)
-		tb.FailNow()
-	}
-}
-
-// ok fails the test if an err is not nil.
-func ok(tb testing.TB, err error) {
-	if err != nil {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf(makeRed+"%s:%d: unexpected error: %s"+makeBlack, filepath.Base(file), line, err.Error())
-		tb.FailNow()
-	}
-}
-
-// equals fails the test if exp is not equal to act.
-func equals(tb testing.TB, exp, act interface{}) {
-	if !reflect.DeepEqual(exp, act) {
-		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf(makeRed+"%s:%d:\n\n\texp: %#v\n\n\tgot: %#v"+makeBlack, filepath.Base(file), line, exp, act)
-		tb.FailNow()
-	}
-}
-
-// The following assertions are inspired by the ok, assert and equals above
-
 // assertSideEffectsContainsMessageWith fails the test instantly if the supplied side effects does not contain a message for the supplied actor with the supplied expected signed state.
 func assertProposalSent(t *testing.T, ses protocols.SideEffects, sp consensus_channel.SignedProposal, to actor) {
-	_, file, line, _ := runtime.Caller(1)
-	if len(ses.MessagesToSend) != 1 {
-		fmt.Printf(makeRed+"%s:%d:\n\n\texpected one message"+makeBlack, filepath.Base(file), line)
-		t.FailNow()
-	}
-	if len(ses.MessagesToSend[0].SignedProposals) != 1 {
-		fmt.Printf(makeRed+"%s:%d:\n\n\texpected one signed proposal"+makeBlack, filepath.Base(file), line)
-		t.FailNow()
-	}
+
+	helpers.Assert(t, len(ses.MessagesToSend) == 1, "expected one message")
```

Comment:
There's a (relatively minor) issue here: one of the benefits of `helpers.Assert` is it points out the line number of the failed assertion. This function would currently print a less useful error to the console: `virtualfund_single_hop_test.go:236: ...`.  This is because `helpers.Assert` looks at one function up the call stack.

Reply:
That's a good point! I think for now I'm going to just ignore it. However we could always look at creating an assert that takes in a call stack depth.